ui/lib/kv/addon/components/page/secret/details.js

```diff
@@ -106,14 +106,59 @@ export default class KvSecretDetails extends Component {
     }
   }
 
-  get isDeactivated() {
-    return this.args.secret.state === 'created' ? false : true;
-  }
-
   get hideHeaders() {
     return this.showJsonView || this.emptyState;
   }
 
+  get versionState() {
+    const { secret, metadata } = this.args;
+    if (secret.failReadErrorCode !== 403) {
+      return secret.state;
+    }
+    // If the user can't read secret data, get the current version
+    // state from metadata versions
+    if (metadata?.sortedVersions) {
+      const version = this.args.version;
```

Comment:
we don't need to pass version in as an arg because we have access to the query params here: 
` this.router.currentRoute.queryParams.version`